src/SchemaRegistry.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+import { Schema } from "./struct/Schema.sol";
+
+contract SchemaRegistry {
+  mapping(bytes32 id => Schema schema) public schemas;
+  uint256 public numberOfSchemas = 0;
+
+  error SchemaAlreadyExists();
+  error SchemaNameMissing();
+  error SchemaStringMissing();
+
+  event SchemaCreated(string name, bytes32 schemaId);
```

Comment:
@EugeniyBykov : from an indexer's perspective: would it be better if the event contained all four schema properties?

Reply:
+1

Reply:
@orbmis 
I dont think it can give any value @jonson